src/main/java/org/spongepowered/api/text/Texts.java

```diff
@@ -233,10 +286,85 @@ public static Text of(Object... objects) {
             }
         }
 
+        if (builder.children.size() == 1) {
+            // Single content, reduce Text depth
+            return builder.children.get(0);
+        }
         return builder.build();
     }
 
     /**
+     * Builds a {@link Text} from a given array of content objects. This method
+     * assumes, that all elements of the array are content and not formatting.
+     *
+     * <p>This will create the correct {@link Text} instance if the input object
+     * is the input for one of the {@link Text} types or convert the object to a
+     * string otherwise.</p>
+     *
+     * @param separator The separator to be inserted between elements
+     * @param objects The array that contains the content
+     * @return The built text object
+     */
+    public static Text joinOf(Text separator, Object... objects) {
+        TextBuilder builder = builder();
+        boolean first = true;
+
+        for (Object part : objects) {
+            if (!first) {
+                builder.append(separator);
+            }
+            first = false;
+            builder.append(Texts.of(part));
+        }
+        return builder.build();
+    }
+
+    /**
+     * Builds a {@link Text} from a given {@link Iterable}. This method
+     * assumes, that all elements of the iterable are content and not
+     * formatting.
+     *
+     * <p>This will create the correct {@link Text} instance if the input object
+     * is the input for one of the {@link Text} types or convert the object to a
+     * string otherwise.</p>
+     *
+     * @param separator The separator to be inserted between elements
+     * @param contents The iterable that contains the content
+     * @return The built text object
+     */
+    public static Text joinOf(Text separator, Iterable<?> contents) {
+        TextBuilder builder = builder();
+        boolean first = true;
+
+        for (Object part : contents) {
+            if (!first) {
+                builder.append(separator);
+            }
+            first = false;
+            builder.append(Texts.of(part));
+        }
+        return builder.build();
+    }
+
+    /**
+     * Creates a new Text instance with all {@link Placeholder}s replaced. All
+     * placeholders without a non-null replacement are ignored. All replacements
+     * will be wrapped in a {@link Text} using {@link Texts#of(Object...)} the
+     * color and the style from the placeholder are transfered to that method as
+     * well. Useful for lazy fetching replacements.
+     *
+     * @param template The template text in which all {@link Placeholder}s
+     *        should be replaced
+     * @param replacements The values available to replace the placeholders
+     * @return The text with all possible placeholders replaced
+     */
+    public static Text format(Text template, Function<String, ?> replacements) {
```

Comment:
With Java8 working there is really no issue in using functions anymore, and since there is no difference implementaion wise why should i prevent it?
I need the function in TextArgument.apply anyway.